src/app/public/auth/mfa/mfa.component.ts

```diff
@@ -0,0 +1,14 @@
+import { Component, Input } from "@angular/core";
+
+@Component({
+    selector: 'awscognito-mfa',
+    templateUrl: './mfa.html'
+})
+export class MFAComponent {
+    @Input() destination: string;
+    @Input() onSubmit: (code: string) => void;
```

Comment:
@joe455 are you referring to this? Not sure why you're reference `submit`; it should be `onSubmit`. This is passed in from login.html #122

Reply:
Yes..When I am using OnSub,i with angular4 I am getting error "OnSubmit is not allowed to bind for security reasons use Submit instead"

Reply:
@hellofornow Got another Issue now.
Once user got registered and not confirmed his phone number. Then, again if he is trying to login I am navigating them to confirmation page(where phone number is not present  so it is undefined). Now in confirmation page the user must enter his phone number but on entering one character the phone number field is disappearing. How can I create work around for this.?

Reply:
Hi @hellofornow ,
 Is MFA will work for only phone_numbers. I want users to choose users where they can receive code(email / phone). Can you shed some lite on this?

Reply:
I just tested w/ latest version of ng (5.1.0).
I don't think you've copied my code properly.
My version has `[onSubmit]` -- lowercase `o`

Reply:
BTW -- I just pushed my changes which includes updating to latest ng, etc.

Reply:
ohh..Sorry.
Usually I wont copy code rather will see the flow and will write my own.Might be missed in this way.
Thanks though!!

Reply:
@hellofornow Can you please help me to extend my app to use custom authentication flow?
Never mind!!! I am quite new to this.